src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java

```diff
@@ -54,35 +49,42 @@ public class IntakeSubsystem extends SubsystemBase {
 
   private DutyCycleEncoder throughBoreEncoder;
 
-  // Sensor closest to birdhouse
-  private TOFSensorSimple sensor;
-  private SimDeviceSim simSensor;
-  private SimBoolean simbeam;
-
   private Mechanism2d mech = new Mechanism2d(2, 2);
   private ComplexWidget widget;
   private MechanismLigament2d pivotArmMech;
 
   private enum IntakeState {
-    Idle,
-    atSensor,
-    coralLoaded
+    FLOOR_INTAKE,
+    FLOOR_OUTTAKE,
+    HP_CORAL_INTAKE,
+    L1_SCORE_MODE_A,
+    L1_SCORE_MODE_B,
+    L1_SCORE,
+    ALGAE_INTAKE,
+    ALGAE_HOLD,
+    ALGAE_OUTTAKE,
+    IDLE,
+    LOCATE_CORAL,
+    PASS_CORAL_TO_SCORER,
+    HOLD_CORAL
   }
 
-  private IntakeState state = IntakeState.Idle;
-  private Timer timer = new Timer();
+  private IntakeState state = IntakeState.IDLE;
+  private Timer t = new Timer();
 
   private SingleJointedArmSim pivotSim;
 
   private DebugEntry<Double> pivotOutput;
   private DebugEntry<String> currentCommand;
 
+  private Sensors sensors;
+
   public IntakeSubsystem() {
     intakeMotor = new TalonFX(CANIDs.kIntakeMotor);
     pivotMotor = new TalonFX(CANIDs.kPivotMotor);
     conveyorMotor = new TalonFX(CANIDs.kConveyorMotor);
-    sensor = new TOFSensorSimple(CANIDs.kConveyorSensor, Inches.of(1), TOFType.LASER_CAN);
     throughBoreEncoder = new DutyCycleEncoder(DIOConstants.kthroughBoreEncoderID, 1, armZero);
+    sensors = new Sensors();
```

Comment:
This won't work to simulate because it is a new instance. You will need to pass in the Sensors instance from RobotContainer for the simulation functions to work correctly.